pkg/installer/packages/fixtures/server.go

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package fixtures contains test datadog package fixtures.
+package fixtures
+
+import (
+	"embed"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/installer/packages/utils"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/stretchr/testify/require"
+)
+
+// Fixture represents a package fixture.
+type Fixture struct {
+	Package     string
+	Version     string
+	layoutPath  string
+	contentPath string
+	configPath  string
+	indexDigest string
+}
+
+var (
+	// FixtureSimpleV1 is a simple package fixture with version v1.
+	FixtureSimpleV1 = Fixture{
+		Package:     "simple",
+		Version:     "v1",
+		layoutPath:  "oci-layout-simple-v1.tar",
+		contentPath: "simple-v1",
+		configPath:  "simple-v1-config",
+	}
+	// FixtureSimpleV2 is a simple package fixture with version v2.
+	FixtureSimpleV2 = Fixture{
+		Package:     "simple",
+		Version:     "v2",
+		layoutPath:  "oci-layout-simple-v2.tar",
+		contentPath: "simple-v2",
+		configPath:  "simple-v2-config",
+	}
+	// FixtureSimpleV1Linux2Amd128 is a simple package fixture with version v1 for linux/amd128.
+	FixtureSimpleV1Linux2Amd128 = Fixture{
+		Package:     "simple",
+		Version:     "v1",
+		layoutPath:  "oci-layout-simple-v1-linux2-amd128.tar",
+		contentPath: "simple-v1",
+	}
+	ociFixtures = []*Fixture{&FixtureSimpleV1, &FixtureSimpleV2, &FixtureSimpleV1Linux2Amd128}
+)
+
+//go:embed *
+var fixturesFS embed.FS
+
+func buildOCIRegistry(t *testing.T) *httptest.Server {
+	s := httptest.NewServer(registry.New())
+	for _, f := range ociFixtures {
+		tmpDir := t.TempDir()
+		file, err := fixturesFS.Open(f.layoutPath)
+		require.NoError(t, err)
+		err = utils.ExtractTarArchive(file, tmpDir, 1<<30)
+		require.NoError(t, err)
+
+		layout, err := layout.FromPath(tmpDir)
+		require.NoError(t, err)
+		index, err := layout.ImageIndex()
+		require.NoError(t, err)
+
+		url, err := url.Parse(s.URL)
+		require.NoError(t, err)
+		src := path.Join(url.Host, f.Package)
+		ref, err := name.ParseReference(src)
+		require.NoError(t, err)
+		err = remote.WriteIndex(ref, index)
+		require.NoError(t, err)
+
+		digest, err := index.Digest()
+		require.NoError(t, err)
+		f.indexDigest = digest.String()
+	}
+	return s
+}
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 Praise:** ‏Really nice